cvat-ui/src/components/annotation-page/canvas/views/canvas2d/canvas-wrapper.tsx

```diff
@@ -483,13 +483,27 @@ class CanvasWrapperComponent extends React.PureComponent<Props> {
         if (prevProps.highlightedConflict !== highlightedConflict) {
             const severity: HighlightSeverity | undefined = highlightedConflict
                 ?.severity as unknown as HighlightSeverity;
-            const highlightedClientIDs = (highlightedConflict?.annotationConflicts || [])
+            const highlightedShapeClientIDs = (highlightedConflict?.annotationConflicts || [])
                 .map((conflict: AnnotationConflict) => annotations
-                    .find((state) => state.serverID === conflict.serverID && state.objectType === conflict.type),
+                    .find((state) => conflict.type !== 'tag' && state.serverID === conflict.serverID && state.objectType === conflict.type),
                 ).filter((state: ObjectState | undefined) => !!state)
                 .map((state) => state?.clientID) as number[];
 
-            canvasInstance.highlight(highlightedClientIDs, severity || null);
+            // if current shape conflict ids is empty then there might be a conflict in tag annotations
+            if (highlightedShapeClientIDs.length === 0) {
+                const highlightedTagClientIDs = (highlightedConflict?.annotationConflicts || [])
+                    .map((conflict: AnnotationConflict) => annotations
+                        .find((state) => conflict.type === 'tag' && state.serverID === conflict.serverID && state.objectType === conflict.type),
+                    ).filter((state: ObjectState | undefined) => !!state)
+                    .map((state) => state?.clientID) as number[];
+
+                const states = annotations.filter((state) => highlightedTagClientIDs.includes(state.clientID || 0));
+                if (states) {
+                    this.updateHighlightedConflict(highlightedConflict);
+                }
```

Comment:
Why do we need this logic? It seems it doesnt do anything useful. The same higlightedConflict is already in the state of the component. I belive we need to remove it

Reply:
True, sorry my bad while editing upon the old logic I didn't considered this.